src/transformers/models/character_bert/tokenization_character_bert.py

```diff
@@ -0,0 +1,926 @@
+        # This is used for converting MLM ids into tokens
+        if mlm_vocab_file is None:
+            self.ids_to_tokens = None
+        else:
+            if not os.path.isfile(mlm_vocab_file):
+                raise ValueError(
+                    f"Can't find a vocabulary file at path '{mlm_vocab_file}'. "
+                    "To load the vocabulary from a pretrained model use "
+                    "`tokenizer = CharacterBertTokenizer.from_pretrained(PRETRAINED_MODEL_NAME)`"
+                )
+            self.ids_to_tokens = build_mlm_ids_to_tokens_mapping(mlm_vocab_file)
+        # Tokenization is handled by BasicTokenizer
+        self.do_basic_tokenize = do_basic_tokenize
+        if do_basic_tokenize:
+            self.basic_tokenizer = BasicTokenizer(
+                do_lower_case=do_lower_case,
+                never_split=never_split,
+                tokenize_chinese_chars=tokenize_chinese_chars,
+                strip_accents=strip_accents,
+            )
+        # Then, a CharacterMapper is responsible for converting tokens into character ids
+        self.max_word_length = max_word_length
+        self._mapper = CharacterMapper(max_word_length=max_word_length)
+
+    def __repr__(self) -> str:
+        # NOTE: we overwrite this because CharacterBERT does not have self.vocab_size
+        return (
+            f"CharacterBertTokenizer(name_or_path='{self.name_or_path}', "
+            + (f"mlm_vocab_size={self.mlm_vocab_size}, " if self.ids_to_tokens else "")
+            + f"model_max_len={self.model_max_length}, is_fast={self.is_fast}, "
+            + f"padding_side='{self.padding_side}', special_tokens={self.special_tokens_map_extended})"
+        )
+
+    def __len__(self):
+        """
+        Size of the full vocabulary with the added tokens.
+        """
+        # return self.vocab_size + len(self.added_tokens_encoder)
+        return 0 + len(self.added_tokens_encoder)
+
+    @property
+    def do_lower_case(self):
+        return self.basic_tokenizer.do_lower_case
+
+    @property
+    def vocab_size(self):
+        raise NotImplementedError("CharacterBERT does not use a token vocabulary.")
+
+    @property
+    def mlm_vocab_size(self):
+        if self.ids_to_tokens is None:
+            raise ValueError(
+                "CharacterBertTokenizer was initialized without a MLM "
+                "vocabulary. You can either pass one manually or load a "
+                "pre-trained tokenizer using: "
+                "`tokenizer = CharacterBertTokenizer.from_pretrained(PRETRAINED_MODEL_NAME)`"
+            )
+        return len(self.ids_to_tokens)
+
+    def add_special_tokens(self, *args, **kwargs):
+        raise NotImplementedError("Adding special tokens is not supported for now.")
+
+    def add_tokens(self, *args, **kwargs):
+        # We don't raise an Exception here to allow for ignoring this step.
+        # Otherwise, many inherited methods would need to be re-implemented...
```

Comment:
Think we should raise here if no tokens can't be added. Which inherited methods would need to be re-implemented?